### DISCUSSION TITLE
does mirth contain xml schemas for exported channels, code templates, alerts and config?

### DISCUSSION THREAD

**Discussion by ppazos** [nextgenhealthcare/connect#5249] [General]
When exporting and importing different mirth connect elements (channels, alerts, config, etc.) between different mirth connect versions, the import works ok when the channel is on the same or previous version at the mcserver. Though importing channels on newer versions to older mcservers doesn't work.

Since everything is exported as xml I'm wondering if mirth provides the xml schemas flirt those elements to be able to validate them before importing so we can do manual changes to the xmls to allow importing newer channels on older mirth servers.

Thanks.

**Comment by siddharthsharma9258** [comment]
You could try your self comparing channel export xml from 3.4 and 3.5 (example) and fill in the gaps. But it would be tedious I believe.

**Reply by ppazos** [reply]
That is what I try to avoid and the key of the question, the idea is to check the channels on their versions against a target XSD on a specific Mirth version to see if there are any validation issues, if there are no breaking changes I can modify the Mirth version number in the channel, alert, code template or config XML and import that into an older Mirth Server.

**Reply by ppazos** [reply]
I'm guessing the channel, alert, code template and configuration XMLs are used in an ad-hoc way in Mirth and there is no formal schema, which is kind of a bummer.

**Comment by jonbartels** [comment]
I don't think MC exposes an XML schema. It's XML data is generated by XStream serializing Java classes. 

To generate a schema a developer would have to identify all the possible serialized classes (for example Channel and then the FileReader connector properties) then build the schema from those classes and fields.

**Reply by ppazos** [reply]
gotcha, I thought they defined their own XML format for each channel/alert/etc. Considering that, I see why it's not possible to downgrade to a previous version.

**Comment by siddharthsharma9258** [comment]
I think OP is looking for a standard way in which any connector or channel data can be expressed, so downgrading is possible.

Could be a useful enhancement request, I think?
